### PR TITLE
Python: Include all assignments in data flow paths

### PR DESCRIPTION
**Pull request by RasmusWL**
Like Ruby did in https://github.com/github/codeql/pull/12566

LHS of assignments will always be included in path-graph. 

```py
x = SOURCE
y = x # <-- would not be part of path-graph before
SINK(x)
```

this is **super** important if there are 100 of lines of code before and after the `y = x` assignment (which I've seen in quite a few real world examples).

## Old behavior

image

## New behavior

image

### CODE REVIEW

**Comment by RasmusWL** [comment]
From looking at the actual paths shown by CodeQL extension, I noticed that for the code below we would force the LHS of both assignments to be part of the path graph, even though the RHS is already part of the path graph...

```py
x = SOURCE
y = identify(x)
SINK(y)
```

Initially I thought this was stupid, and wanted to get rid of it. However, after using it for a bit, I actually found it really nice! It made everything so obvious and did not force me to think... when seeing RHS of an assignment, there is no need to remember the name of the LHS, I can just follow it step-by-step :tada:

So in conclusion, I actually think this is a really nice feature!

no I don't think I will

**Comment by RasmusWL** [comment]
Performance looks fine as well :+1:

**Review by turbo — DISMISSED** [review]

**Review by yoff — APPROVED** [review]
Only a typo. Great test setup 👍 
Very happy to see path explanations get some love ❤️

**Review by yoff — APPROVED** [review]
LGTM
